hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -2432,7 +2432,7 @@ public static HoodieData<HoodieRecord> convertFilesToPartitionStatsRecords(Hoodi
     }
     Lazy<Option<Schema>> lazyWriterSchemaOpt = writerSchemaOpt.isPresent() ? Lazy.eagerly(writerSchemaOpt) : Lazy.lazily(() -> tryResolveSchemaForTable(dataTableMetaClient));
     final List<String> columnsToIndex = getColumnsToIndex(dataTableMetaClient.getTableConfig(), metadataConfig, Either.right(lazyWriterSchemaOpt),
-        dataTableMetaClient.getActiveTimeline().filterCompletedInstants().empty(), recordTypeOpt);
+        dataTableMetaClient.getActiveTimeline().getWriteTimeline().filterCompletedInstants().empty(), recordTypeOpt);
```

Comment:
any completed action should be sufficient to tell whether table is initialized or not, right.. why `getWriteTimeline` specifically? Do we just want to prune the number of instants for empty() check, or was there some bug?

Reply:
for a table containing just 1 entry in timeline and its a completed rollback instant, we dont' won't be able to parse the schema. hence this change.